### PR TITLE
Improve parsing resilience to whitespace, enforce des/ as last arg, add non-finite amount checks

### LINKED ISSUE #145: Description field can absorb compulsory fields if declared first
Solution: strictly enforce the optional description field as the last parameter, while allowing compulsory fields to be in any order.

### LINKED ISSUE #146: Adding, modifying and budgeting do not have non-finite checks

### PR DESCRIPTION
**Pull request by sanjay-shiva-kumar**
Fixes #145,
fixes #146

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ Tip: Type `help` after launch to see every available command.
     - `c/` for category (single word or phrase without leading/trailing spaces).
     - `d/` for date in `YYYY-MM-DD` format.
     - `des/` for an optional description. If omitted, the entry has no description.
-- Parameters can be input in **any order**. For example, if `add-expense` requires the `a/<amount>`, `c/<category>` and `d/<YYYY-MM-DD>` parameters, it can be input in any order (e.g. `c/<category>`, `a/<amount>`, `d/<YYYY-MM-DD>`).
+- Compulsory parameters can be input in **any order**. For example, if `add-expense` requires the `a/<amount>`, `c/<category>` and `d/<YYYY-MM-DD>` parameters, it can be input in any order (e.g. `c/<category>`, `a/<amount>`, `d/<YYYY-MM-DD>`).
+  - However, if you include the optional description (`des/`), place it after all other parameters—everything after `des/` is treated as part of the description.
```
Comment (duckyfuz) [inline docs/UserGuide.md]: LGTM! This is an excellent clarification.

The original documentation was misleading by stating all parameters could be in any order. This change correctly distinguishes between the compulsory parameters (which remain flexible) and the optional des/ parameter, which has a specific constraint.

Adding the rule that des/ must be placed last—and explaining why ("everything after des/ is treated as part of the description")—is a crucial improvement. This makes the documentation more accurate and will prevent user confusion and parsing errors.

File: src/test/java/seedu/fintrack/ParserTest.java [hunk 2]
```diff
@@ -360,6 +460,91 @@ public void parseModifyExpense_invalidParams_throws() {
...
+                + Ui.DESCRIPTION_PREFIX + "tab spacing";
+
+        var result = Parser.parseModifyExpense(input);
+        assertEquals(1, result.getKey());
+        Expense e = result.getValue();
+        assertEquals(42.0, e.getAmount(), 1e-9);
+        assertEquals(ExpenseCategory.FOOD, e.getCategory());
+        assertEquals(LocalDate.of(2025, 10, 7), e.getDate());
+        assertEquals("tab spacing", e.getDescription());
+    }
```
Comment (duckyfuz) [inline src/test/java/seedu/fintrack/ParserTest.java]: Great that you've spotted this edge case

File: src/test/java/seedu/fintrack/ParserTest.java [hunk 3]
```diff
@@ -590,6 +861,107 @@ public void parseAddIncome_invalidAmountOrDate_throws() {
...
+                + Ui.AMOUNT_PREFIX + "88.8 "
+                + Ui.DESCRIPTION_PREFIX + "Reordered prefixes ";
+
+
+        Income income = Parser.parseAddIncome(input);
+        assertEquals(88.8, income.getAmount(), 1e-9);
+        assertEquals(IncomeCategory.GIFT, income.getCategory());
+        assertEquals(LocalDate.of(2025, 12, 24), income.getDate());
+        assertEquals("Reordered prefixes", income.getDescription());
+    }
```
Comment (duckyfuz) [inline src/test/java/seedu/fintrack/ParserTest.java]: Good happy path chosen here
